plugin/src/functionalTest/kotlin/io/qalipsis/gradle/cloud/QalipsisCloudPluginFunctionalTest.kt

```diff
@@ -0,0 +1,861 @@
+                    java
+                }
+            """.trimIndent()
+        )
+        mockServerClient.`when`(
+            request()
+                .withPath("/scenarios")
+                .withMethod("GET")
+        ).respond(
+            response()
+                .withStatusCode(statusCode)
+        )
+
+        // When
+        val runner = GradleRunner.create()
+            .forwardOutput()
+            .withPluginClasspath()
+            .withProjectDir(tempFolder.toFile())
+            .withDebug(true)
+            .withArguments()
+        val result = runner.buildAndFail()
+
+        // Then
+        result.output shouldContain """
+           There was an authentication issue, please check the API token you configured as Gradle property `qalipsis.cloud.registry.token`. Is it still valid and has the permission to read scenarios?
+        """.trimIndent()
+
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = [502, 504])
+    fun `should return the right error message when there is a server error`(
+        statusCode: Int,
+        @TempDir tempFolder: Path
+    ) {
+        // Given
+        val propertiesFile = tempFolder.resolve("gradle.properties")
+        propertiesFile.writeText(
+            """
+            qalipsis.cloud.registry.url=http://localhost:${mockPort}/scenarios
+            qalipsis.cloud.registry.list=true
+            qalipsis.cloud.registry.token=my-token
+        """.trimIndent()
+        )
+        val buildFile = tempFolder.resolve("build.gradle.kts")
+        buildFile.writeText(
+            """
+                plugins {
+                    id("io.qalipsis.cloud")
+                    java
+                }
+            """.trimIndent()
+        )
+        mockServerClient.`when`(
+            request()
+                .withPath("/scenarios")
+                .withMethod("GET")
+                .withHeader(Header("Authorization", "Bearer my-token"))
+        ).respond(
+            response()
+                .withStatusCode(statusCode)
+                .withBody("Gateway Timeout")
+        )
+
+        // When
+        val runner = GradleRunner.create()
+            .forwardOutput()
+            .withPluginClasspath()
+            .withProjectDir(tempFolder.toFile())
+            .withDebug(true)
+            .withArguments("tasks", "--all")
+        val result = runner.build()
+
+        // Then
+        result.output shouldContain """
+            The QALIPSIS server is currently not available, and the list of existing scenarios cannot be retrieved.
+             Please try again in a moment.
+             If the problem persists, please contact the support.
+       """.trimIndent()
+        assertTrue(result.output.contains("Qalipsis cloud tasks"))
+        assertTrue(result.output.contains("qalipsisPublish"))
+    }
+
+    @Test
+    fun `should return the right error message for when any other error occurs while fetching scenarios`(@TempDir tempFolder: Path) {
```

Comment:
The function name does not tell if it is about listing, deleting or publishing scenarios. Splitting the class would definitely make readability better.